pythonmonkey_require.py

```diff
@@ -97,17 +102,29 @@
 pm.eval("python.paths = python.paths.split(':'); true"); # fix when pm supports arrays
 
 # Implement enough of require('fs') so that ctx-module can find/load files
-def statSync_inner(filename):
+
+def statSync_inner(filename: str) -> Union[Dict[str, int], bool]:
+    """
+    Inner function for statSync.
+
+    Returns:
+        Union[Dict[str, int], False]: The mode of the file or False if the file doesn't exist.
+    """
     from os import stat
     if (path.exists(filename)):
         sb = stat(filename)
         return { 'mode': sb.st_mode }
     else:
         return False
 
-def readFileSync(filename, charset):
-    fileHnd = open(filename, "r")
-    return fileHnd.read()
+def readFileSync(filename, charset) -> str:
+    """
+    Utility function for reading files.
+    Returns:
+        str: The contents of the file
+    """
+    with open(filename, "r") as fileHnd:
+        return fileHnd.read()
```

Comment:
You mentioned this morning that Python doesn't auto-close files when their handles fall out of scope?

If that's true, then readFileSync needs to close fileHnd before returning its contents

Also, I was going to make this change but merge conflicts......... I think the open() call should include `encoding=charset`

Reply:
Got it! I'll make that change for the `encoding=charset`.

Re: the `with open()...` python is a little weird. If one just uses `open`, the `close` is never called, but if one uses `with open` according to [PEP 343](https://peps.python.org/pep-0343/), once we fall out of scope of the `with` (or we return), the `__exit__` of the object in the `with` block is called. Thankfully, the `__exit__` for  the file object calls the `close`.